### PR TITLE
update and fix icescrum service

### PR DESCRIPTION
**Pull request by vbarrier**
- Fix bug with access_token
- update tests
- update doc

### CODE REVIEW

**Comment by vbarrier** [comment]
Hi @bswinnerton Thank you for your comment. I didn't notice the errors! They have been fixed :)
Thank you for your review :)

File: lib/services/icescrum.rb [hunk 1]
```diff
@@ -1,26 +1,48 @@
...
-	username = data['username'].to_s.gsub(/\s+/, "")
-	project_key = data['project_key'].to_s.upcase.gsub(/\s+/, "")
-	password = data['password'].to_s.gsub(/\s+/, "")
+	# Cloud only support access token
+	if data['base_url'].to_s.empty?
+		raise_config_error "Access token mandatory for cloud" if data['access_token'].to_s.empty?
+	end
 
+	 # Support old authentification for R6
+	if data['access_token'].to_s.empty?
```
Comment (vbarrier) [inline lib/services/icescrum.rb]: Thank you

File: lib/services/icescrum.rb [hunk 2]
```diff
@@ -1,26 +1,48 @@
...
-	raise_config_error "Invalid username" if data['username'].to_s.empty?
-	raise_config_error "Invalid password" if data['password'].to_s.empty?
-	raise_config_error "Invalid project key" if data['project_key'].to_s.empty?
-
-	username = data['username'].to_s.gsub(/\s+/, "")
-	project_key = data['project_key'].to_s.upcase.gsub(/\s+/, "")
-	password = data['password'].to_s.gsub(/\s+/, "")
+	# Cloud only support access token
+	if data['base_url'].to_s.empty?
+		raise_config_error "Access token mandatory for cloud" if data['access_token'].to_s.empty?
```
Comment (vbarrier) [inline lib/services/icescrum.rb]: Thank you

File: test/icescrum_test.rb [hunk 3]
```diff
@@ -46,37 +53,33 @@ def test_push_valid_custom_url
     @stubs.verify_stubbed_calls
   end
 
-  def test_push_lowcase_project_key
```
Comment (vbarrier) [inline test/icescrum_test.rb]: True

**Comment by vbarrier** [comment]
H @bswinnerton, 

You are right, it's not only a bug fix pull request but it is also intended to maintain our Cloud service. So I thought it was okay to also add our new (and only) access token version.  

I'm already working on an app that will use your API for the futur. I have also seen that we can prefill the new personal token form using a GET request.

Thank you :)

**Comment by vbarrier** [comment]
Hello guys!

Any new about this pull request ready to be merged :) ?

**Comment by gjtorikian** [comment]
Really sorry for the late reply on this. 

Thanks for this PR! Currently this project is [slated for deprecation](https://developer.github.com/changes/2018-04-25-github-services-deprecation/) and is only accepting critical bugfixes. 

Thank you for your time and understanding, but I will have to close this out. 💞

**Pull request closed without merging**